plugins/GSdx/Renderers/DX11/GSDevice11.cpp

```diff
@@ -632,14 +638,17 @@ void GSDevice11::ClearStencil(GSTexture* t, uint8 c)
 
 GSTexture* GSDevice11::CreateSurface(int type, int w, int h, int format)
 {
+	ASSERT(w > 0 && w <= m_d3d_texsize);
+	ASSERT(h > 0 && h <= m_d3d_texsize);
+
 	HRESULT hr;
 
 	D3D11_TEXTURE2D_DESC desc;
 
 	memset(&desc, 0, sizeof(desc));
 
-	desc.Width = std::max(1, std::min(w, 8192)); // Texture limit for D3D10 min 1, max 8192
-	desc.Height = std::max(1, std::min(h, 8192));
+	desc.Width = std::max(1, std::min(w, m_d3d_texsize)); // Texture limit for D3D10 min 1, max 8192 D3D10, max 16384 D3D11.
+	desc.Height = std::max(1, std::min(h, m_d3d_texsize));
```

Comment:
Maybe  use ``std::clamp``instead?

Reply:
It's a C++17 feature so would probably cut off compilers - if anything it should be made in code somewhere.

Reply:
Yeah, can't use clamp over here (vs 2015). Buildbot and appveyor also use 2015.

Reply:
2015.3 should [actually support](https://docs.microsoft.com/en-us/cpp/overview/visual-cpp-language-conformance?view=vs-2015) clamp.

Reply:
Yeah there is a switch to use c++17 but I'll leave it out. Plans to switch to c++17 are set for after 1.6 release.